spinnaker/deck
# feat(provider/azure): Add userAssignedIdentities to azure UI

In order for Azure virtual machines to access other Azure resources, they need an identity to be associated with the VM. This change sets the associated identity on the VM. This is already supported by clouddriver, just needs to be exposed on the UI.

**New UI element**:

![image](https://user-images.githubusercontent.com/910168/131415951-72c15c7e-d0e4-40ab-8908-3e076269a721.png)


**Testing done**:

1. Add user-assigned identity value
2. Deploy scale set
3. Verify that user-assigned identity was assigned to the scale set instance:

```
    "identity": {
        "type": "UserAssigned",
        "userAssignedIdentities": {
            "/subscriptions/<SUBSCRIPTION_ID>/resourceGroups/fooservice-centralus/providers/Microsoft.ManagedIdentity/userAssignedIdentities/my-identity": {
                "principalId": "REDACTED",
                "clientId": "REDACTED"
            }
        }
    }
```

4. Remove user-assigned identity value
5. Deploy scale set
6. Verify that user-assigned identity was **not** assigned to the scale set instance.
